tests/core/p2p-proto/test_sync.py

```diff
@@ -87,6 +91,25 @@ def chaindb_with_gaps(chaindb_fresh, chaindb_1000):
     yield chaindb_fresh
 
 
+@pytest.fixture
+def chaindb_with_block_gaps(chaindb_fresh, chaindb_1000):
+    # Make a chain with gaps. This fixture can not be used in a test alongside `chaindb_fresh`
+    # because it alters the `chaindb_fresh` fixture.
+
+    for block_number in range(1, 1001):
+        header_at = chaindb_1000.get_canonical_block_header_by_number(block_number)
+        chaindb_fresh.persist_header_chain([header_at])
```

Comment:
Silly little thing, but I this should run faster as:
```suggestion
    all_headers = [
        chaindb_1000.get_canonical_block_header_by_number(block_number)
        for block_number in range(1, 1001)
    ]
    chaindb_fresh.persist_header_chain(all_headers)
```
(and I think be a tiny bit faster to scan what it's doing)

Reply:
:+1: